### PR TITLE
Add Intel SGX Detection and Enablement

### PR DESCRIPTION
**Pull request by danintel**
- Add how to detect Intel SGX supported
- Mention Intel SGX must be enabled in BIOS

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

File: PREREQUISITES.md [hunk 1]
```diff
@@ -174,8 +174,10 @@ install the Intel SGX driver for both standalone and docker builds.
 You need to install the Intel SGX driver whether you build Avalon standalone
 or using Docker.
 
-First install this package:
+Verify your processor supports Intel SGX with `cpuid | grep SGX:` .
+Also verify Intel SGX is enabled in BIOS.
```
Comment (manojgop) [inline PREREQUISITES.md]: How about giving the link to this video https://www.youtube.com/watch?v=bca5NcjoEdc

File: PREREQUISITES.md [hunk 2]
```diff
@@ -418,6 +422,9 @@ problems.
   you are using an old version of OpenSSL and need to clone the
   `openssl_1.1.1` branch of `intel-sgx-ssl` in the step above
 
+- If the message  `intel_sgx: SGX is not enabled` appears in `/var/log/syslog`
+  Intel SGX needs to be enabled in BIOS
```
Comment (manojgop) [inline PREREQUISITES.md]: Is this a standard recommended way to detect and enable SGX ?

Comment (danintel) [reply PREREQUISITES.md]: No. It is only how to detect in Linux if SGX is enabled in BIOS.

This message appears even if SGX is supported by the processor an even if drivers are installed.

File: PREREQUISITES.md (hunk 1 above)
Comment (danintel) [reply PREREQUISITES.md]: I like John's SGX videos but I do not want to require watching a video to install SGX or Avalon.

Comment (manojgop) [reply PREREQUISITES.md]: What I meant is it's not clear how to verify if Intel SGX is enabled in BIOS ? or is this not part of documentation scope ?

Comment (dcmiddle) [reply PREREQUISITES.md]: cpuid may not be installed on all distros. For example, I don't think it's part of the ubuntu docker image Avalon uses. 
`cat /proc/cpuinfo | grep sgx` 
might be more portable.

**Pull request merged**